lib/test.d.ts

```diff
@@ -1,119 +1,243 @@
+   * subtests, it will be Infinity tests in parallel. If falsy, it would only
+   * run one test at a time. If unspecified, subtests inherit this value from
+   * their parent.
+   *
+   * @default false
    */
   concurrency?: boolean | number;
 
   /**
-   * If truthy, the test is skipped. If a string is provided, that string is displayed in the test results as the reason for skipping the test.
-   * Default: false.
+   * If truthy, and the test context is configured to run only tests, then this
+   * test will be run. Otherwise, the test is skipped.
+   *
+   * @default false
+   */
+  only?: boolean;
+
+  /**
+   * If truthy, the test is skipped. If a string is provided, that string is
+   * displayed in the test results as the reason for skipping the test.
+   *
+   * @default false
    */
   skip?: boolean | string;
 
   /**
-   * If truthy, the test marked as TODO. If a string is provided, that string is displayed in the test results as the reason why the test is TODO.
-   * Default: false.
+   * If truthy, the test marked as TODO. If a string is provided, that string is
+   * displayed in the test results as the reason why the test is TODO.
+   *
+   * @default false
    */
   todo?: boolean | string;
 
   /**
-   * A number of milliseconds the test will fail after. If unspecified, subtests inherit this value from their parent.
-   * Default: Infinity
+   * A number of milliseconds the test will fail after. If unspecified, subtests
+   * inherit this value from their parent.
+   *
+   * @default Infinity
    */
   timeout?: number;
 
-  /**
-   * Allows aborting an in-progress test
-   */
+  /** Allows aborting an in-progress test */
   signal?: AbortSignal;
 }
 
-type TestFn = (t: TestContext) => any | Promise<any>;
+type TestFn = (t: TestContext, done: (result?: any) => void) => any;
 
 export default test;
 
-export function test(name: string, options: TestOptions, fn: TestFn): void;
-export function test(name: string, fn: TestFn): void;
-export function test(options: TestOptions, fn: TestFn): void;
-export function test(fn: TestFn): void;
+/**
+ * @param name The name of the test, which is displayed when reporting test
+ *   results. Default: The name property of fn, or '<anonymous>' if fn does not
+ *   have a name.
+ * @param options Configuration options for the test.
+ * @param fn The function under test. The first argument to this function is a
+ *   TestContext object. If the test uses callbacks, the callback function is
+ *   passed as the second argument. Default: A no-op function.
+ * @returns A {@link Promise} resolved with `undefined` once the test completes.
+ */
+export function test(
+  name?: string,
+  options?: TestOptions,
+  fn?: TestFn
+): Promise<void>;
+export function test(name?: string, fn?: TestFn): Promise<void>;
+export function test(options?: TestOptions, fn?: TestFn): Promise<void>;
+export function test(fn?: TestFn): Promise<void>;
```

Comment:
Those are not optional in the overrides, otherwise TS won't know what signature to match with the user code.
```suggestion
export function test(name: string, fn: TestFn): Promise<void>;
export function test(options: TestOptions, fn?: TestFn): Promise<void>;
export function test(fn: TestFn): Promise<void>;
```

Reply:
They're not strictly necessary for the typechecker to correctly apply the overloads, but it makes sense to me, added back with https://github.com/nodejs/node-core-test/pull/42/commits/9225f9ee9577cd0b8778c9f4664b0564a74fe53a